fontir/src/glyph.rs

```diff
@@ -168,6 +171,101 @@ fn components(
         .collect()
 }
 
+// Operations performed on glyphs with mixed contours/components
+#[derive(Clone, Copy, PartialEq, Eq)]
+enum GlyphOp {
+    // convert comonents in this mixed glyph into contours
+    ConvertToContour,
+    // move contours in this mixed glyph into components
+    MoveContoursToComponent,
+}
+
+/// Fix glyphs with mixed components/contours.
+///
+/// If there are component relationships between these glyphs the leaves have
+/// to be handled first.
+fn resolve_inconsistencies(
+    context: &Context,
+    todo: Vec<(GlyphOp, Arc<Glyph>)>,
+    order: &mut GlyphOrder,
+) -> Result<(), BadGlyph> {
+    let items = todo
+        .into_iter()
+        .map(|(op, glyph)| (glyph.name.clone(), (op, glyph)))
+        .collect::<HashMap<_, _>>();
+
+    for name in component_ordering_within_glyphs(items.values().map(|(_, g)| g.clone())) {
+        let (op, glyph) = items.get(&name).unwrap();
+        match op {
```

Comment:
Nit: I feel like `op.execute(context, glyph)` might be nicer than matching on op to call a fn?